feature/google/src/main/java/com/revenuecat/purchases/google/storeProductConversions.kt

```diff
@@ -48,7 +50,10 @@ fun List<ProductDetails>.toStoreProducts(): List<StoreProduct> {
         basePlans.takeUnless { it.isEmpty() }?.forEach { basePlan ->
             val basePlanBillingPeriod = basePlan.subscriptionBillingPeriod
             val offerDetailsForBasePlan = offerDetailsBySubPeriod[basePlanBillingPeriod] ?: emptyList()
-            productDetails.toStoreProduct(basePlanBillingPeriod, offerDetailsForBasePlan).let { storeProducts.add(it) }
+            val defaultOfferDetails = basePlan // TODOBC5: Change logic here for default offer.
```

Comment:
I feel we are overusing extension functions for this. I think once we add some more complex logic here, we should move this to a factory class that can be better tested on its own.

Reply:
depends on the logic, but if defaultOffer only needs `StoreProduct` info to be calculated, perhaps we can make it a calculated property on storeProduct. but i know we don't know that yet...